src/components/Gists.js

```diff
@@ -1,105 +1,89 @@
-        return res.json();
-      }
-      this.setState({ message: 'No gists found' });
-      return Promise.reject();
-    }).then((data) => {
-      // Load gist data
-      let gists = {};
-
-      // Create list of gists
-      if (data.length > 0) {
-        data.forEach(gist => {
-          let files = [];
-          // Create list of files for each gist
-          Object.values(gist.files).forEach(file => {
-            let type = 'Unknown';
+    this.fetchGistData();
+  }
 
-            // Parse file type
-            switch(file.type) {
-              case TEXT:
-                type = "Text";
-                break;
-              case PYTHON:
-                type = "Python";
-                break;
-              case XML:
-                type = "XML";
-                break;
-            }
+  fetchGistData() {
+    let userId = _.get(this.props, 'match.params.user_id');
+    let url = getUserGistsUrl(userId, 0, 10);
+    
+    fetch(url)
+      .then(res => {
+        // Check status
+        if (res.status >= 200 && res.status < 300) {
+          return res.json();
+        }
+        this.setState({ emptyMessage: 'Invalid username' });
+        return Promise.reject();
+      })
+      .then(data => {
+        if (data === undefined) {
+          console.log('Returned data is undefined');
+          return Promise.reject();
+        }
 
-            files.push({
-              filename: file.filename,
-              url: file.raw_url,
-              type: type
-            });
+        // Create a parsed list of gists
+        let gists = _.map(data, gist => {
+          // Create a new list of truncated file data for each gist
+          let files = _.map(gist.files, file => {
+            let type = _.get(fileMap, file.type, 'Unknown'); // Parsing file type
+            return { filename: file.filename, url: file.raw_url, type}; // Returning updated file object
           });
 
-          gists[gist.id] = {
-            ...gist,
-            username: gist.owner.login,
-            forks: [],
-            files: files,
-          }
+          // Returning updated gist data
+          return { ...gist, username: _.get(gist, 'owner.login', 'N/A'), forks: [], files };
         });
-      } else {
-        this.setState({ message: 'No gists found' });
-      }
-      console.log(gists);
-      this.setState({ gists: gists });
+        console.log(gists);
 
-      // FETCH FORKS //
+        this.setState(gists.length > 0 ? { gists } : { emptyMessage: 'No gists found' });
 
-      Object.values(gists).forEach(gist => {
-        fetch(gist.forks_url).then(res => res.json())
-          .then(data => {
-            console.log('Forks: ', data);
-            gists[gist.id].forks = data;
-            this.setState({ gists });
-          });
+        this.fetchForksData();
+
+        return Promise.resolve();
       });
+  }
 
-      return Promise.resolve();
-    })
-    .catch((res) => {
-      console.log(res);
+  fetchForksData() {
```

Comment:
I am not satisfied with this approach. My gut feeling is that this code should not be in this file.